.github/workflows/ci.yml

```diff
@@ -291,6 +291,34 @@ jobs:
           path: ${{ env.RUST_BIN_DIR }}/${{ matrix.binary }}
           retention-days: 1
 
+  cargo-vet:
+    name: Vet Dependencies
+    runs-on: ubuntu-latest
+    env:
+      CARGO_VET_VERSION: 0.3.0
```

Comment:
Do we need to run in PR or can be just run in staging?

Reply:
Yeah, the goal is to vet dependencies whenever a PR is opened so we notice before merging if theres any warning.
Im taking in consideration [this scenario](https://mozilla.github.io/cargo-vet/how-it-works.html#adding-new-third-party-code).

Reply:
I am concerned that this will slow down builds. I proposed that only performs in the staging

Reply:
Perhaps the suggestion on [this comment](https://github.com/infinyon/fluvio/pull/2795#discussion_r1019649506) helps reduce the scope to `staging` only.